src/rust/engine/src/context.rs

```diff
@@ -123,28 +123,36 @@ impl Core {
       })
       .unwrap_or_else(|e| panic!("Could not initialize Store: {:?}", e));
 
-    let command_runner = match &remote_execution_server {
-      Some(ref address) if remote_execution => BoundedCommandRunner::new(
-        Box::new(process_execution::remote::CommandRunner::new(
-          address,
-          remote_execution_process_cache_namespace.clone(),
-          remote_instance_name.clone(),
-          root_ca_certs.clone(),
-          oauth_bearer_token.clone(),
-          remote_execution_extra_platform_properties.clone(),
-          store.clone(),
-        )),
-        process_execution_remote_parallelism,
-      ),
-      _ => BoundedCommandRunner::new(
+      let mut command_runner: Box<dyn process_execution::CommandRunner> = Box::new(BoundedCommandRunner::new(
         Box::new(process_execution::local::CommandRunner::new(
           store.clone(),
           executor.clone(),
           work_dir.clone(),
           process_execution_cleanup_local_dirs,
         )),
         process_execution_local_parallelism,
-      ),
+      ));
+
+    command_runner = match &remote_execution_server {
```

Comment:
So, while the hardcoded timeout is probably OK for the MVP, I think that we do need a ternary option for enabling speculation before landing this. Something like:
```
--remote-execution-speculation={local_first,remote_first,none}
```

And `local_first` would be a reasonable default. An example ternary option is: [here](https://github.com/pantsbuild/pants/blob/d53985580f9f015a1ed691d289b393e7622649e3/src/python/pants/option/global_options.py#L114-L115).

Reply:
RE. our convo and daniels suggestion
>I'd probably revert the context.rs changes and land them separately (with whatever flags we need).

I want to land the speculating command runner and then follow up with a PR that wires it into the context and adds the needed flags to configure speculation timeout and ternary speculation modes.

Reply:
Works for me!